### PR TITLE
Update BB tests' configurations' code

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2217 
Merge after #2229 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [ ] Have you successfully tested your changes locally? Elaborate:
    > NOT tested yet. BB tests need to be run.
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: envs/monkey_zoo/blackbox/test_configurations/utils.py [hunk 1]
```diff
@@ -1,152 +1,78 @@
...
-        agent_configuration.propagation.exploitation,
-        brute_force=brute_force,
-        vulnerability=vulnerability,
-    )
-    return replace_exploitation_configuration(agent_configuration, exploitation_configuration)
+
+    agent_configuration.propagation.exploitation.brute_force = brute_force
+    agent_configuration.propagation.exploitation.vulnerability = vulnerability
+
+    return agent_configuration
```
Comment (mssalvatore) [inline envs/monkey_zoo/blackbox/test_configurations/utils.py:15]: I don't think we need any of these functions anymore. We can just manipulate the configurations directly in the config files. IMO, these no longer add clarity or simplicity.

Comment (shreyamalviya) [reply envs/monkey_zoo/blackbox/test_configurations/utils.py:15]: All the places where the configuration needs to modified use these functions. I think that's better than specifying the whole thing in multiple places which is repetitive. Plus, if we ever modify the configuration structure, we'll only have to change this in one place.

File: envs/monkey_zoo/blackbox/island_client/monkey_island_client.py [hunk 2]
```diff
@@ -51,7 +50,7 @@ def _set_island_mode(self):
     def _import_config(self, test_configuration: TestConfiguration):
         response = self.requests.put_json(
             "api/agent-configuration",
-            json=AgentConfiguration.to_mapping(test_configuration.agent_configuration),
+            json=test_configuration.agent_configuration.dict(),
```
Comment (mssalvatore) [inline envs/monkey_zoo/blackbox/island_client/monkey_island_client.py:53]: Do we need `simplify=True`?

File: envs/monkey_zoo/blackbox/test_configurations/utils.py (hunk 1 above)
Comment (mssalvatore) [reply envs/monkey_zoo/blackbox/test_configurations/utils.py:15]: If you want to leave them, then they should either make a copy and return the modified copy, or not return anything and modify the input.

Comment (shreyamalviya) [reply envs/monkey_zoo/blackbox/test_configurations/utils.py]: Updated the code to make a copy and return the modified copy.

File: envs/monkey_zoo/blackbox/island_client/monkey_island_client.py (hunk 2 above)
Comment (shreyamalviya) [reply envs/monkey_zoo/blackbox/island_client/monkey_island_client.py:53]: Added

**Review by mssalvatore — APPROVED** [review]
Approved, but test with #2229.

**Comment by shreyamalviya** [comment]
Blackbox tests pass.

**Pull request merged**